Add option to develop Probot on Glitch

dhcodes opened the pull request:
Created a Glitch project with instructions for setting up a new Probot development to make it easier for new users.

Closes #241

anglinb commented:
👋 Hi @dhcodes! Thanks for opening this up--I think Glitch is an awesome way to help people get started!

I checked out the link you added and it dropped me into your project which means I still need to click `Remix` before I can start editing. Would you consider changing the link to https://glitch.com/edit/#!/remix/crystal-colony ? That way when you click the link Glitch will automatically create a new "remix" of your project.

JasonEtco commented:
This is dope <3 Would you be able to change the Glitch template to use Jest instead of Mocha/Expect, so that its consistent with `create-probot-app`'s output?

cc @gr2m, master of Glitch.

dhcodes commented:
Thanks all, and yes, I can fix both of those things up. Will let you know when it's ready.

gr2m commented:
maybe rename https://glitch.com/edit/#!/crystal-colony to something more official looking, like https://glitch.com/edit/#!/probot-hello-world and invite some folks from the Probot team to it in case something goes wrong and you are not around?

bkeepers commented:
👍 this is awesome. I would love to see this linked to from the [Hello World](https://probot.github.io/docs/hello-world/) docs.

dhcodes commented:
Okay, I think I got everything. I used the most recent package.json with jest and other newer packages. I added your logo as well. You may want to update the docs to be Jest-oriented down the road since they are currently mocha.

For now I updated the link to be a remix link and removed the step of clicking the "Remix this" button. The only downside I can see to this approach is users won't see the cool logo but :man_shrugging: 

Main project:
https://glitch.com/edit/#!/probot-hello-world

Collab link (you may want to remove this once team members have joined):
https://glitch.com/edit/#!/join/c4798b26-9c91-40d6-b6de-7ad1649d07c0

This was a fun Hacktoberfest project.

dhcodes commented:
Should I put it in both places or just the hello world?

bkeepers commented:
> Should I put it in both places or just the hello world?

My preference would be hello world.

tbarn commented:
I just wanted to bring up the use of `.data/private-key.pem`. In the past we found we wouldn't be able to use it for the private key since it wasn't actually meant to hide data like `.env`. Has there been a change to this in Glitch since then? 

I've seen some weird behavior with it today, like the file disappearing for me, so then I am not sure if it is there or not.

dhcodes commented:
@tbarn If you put a private key in that file (ex.
```
-----BEGIN ENCRYPTED PRIVATE KEY-----
MIIFDjBABgkqhkiG9w0BBQ0wMzAbBgkqhkiG9w0BBQwwDg
MBQGCCqGSIb3DQMHBAgD1kGN4ZslJgSCBMi1xk9jhlPxPc
9g73NQbtqZwI+9X5OhpSg/2ALxlCCjbqvzgSu8gfFZ4yo+
blahblahblahblah
X0R+meOaudPTBxoSgCCM51poFgaqt4l6VlTN4FRpj+c/Wc
blK948UAda/bWVmZjXfY4Tztah0CuqlAldOQBzu8TwE7WD
H0ga/iLNvWYexG7FHLRiq5hTj0g9mUPEbeTXuPtOkTEb/0
GEs=
-----END ENCRYPTED PRIVATE KEY-----
```

then it won't let non-collaborators see the contents. Instead it says "This is a private file" per your screenshot. Why would that be a bad thing? It isn't sharing the key. There are instructions in the readme that explain how a user can get their own key.

tbarn commented:
@dhcodes Sorry if I wasn't clear. It seems to be working better than a few months ago. 

Still, I've seen the file disappear on the project I created it on and the overall behavior seems to have changed multiple times now in the last few months.

A few months ago in the Glitch forums, it was said that the .data/ folder was not really made for private keys since the .env functionality existed and was more dependable. I'm just not sure if I trust it yet, so I'll probably try to stick with the .env way of going about it.

gr2m commented:
> I just wanted to bring up the use of .data/private-key.pem. In the past we found we wouldn't be able to use it for the private key since it wasn't actually meant to hide data like .env. Has there been a change to this in Glitch since then?

I checked in about the `.data` folder a while ago, you are safe putting sensitive data in there, it’s meant to be used for storing user data, for example.

> I've seen the file disappear

It is still there, open console (in advanced options) and type `ls .data`, you should see the file in there

dhcodes commented:
You can close this one. I opened a new PR #328 to add this to Hello World. I would've done it on the same PR but I was lazily using the web editor and am not sure how to edit multiple files in one PR using it.

Thanks and sorry for the delay.

j-f1 commented:
> [I’]m not sure how to edit multiple files in one PR using it.

If you click the `patch-1` link below this comment, you’ll be able to browse the branch this PR is coming from and directly edit the files in it from the web editor.

dhcodes commented:
Thanks @j-f1! Updated this PR
